### PR TITLE
New WGHATickets Spice

### PR DESCRIPTION
**Pull request by bigcurl**
# Spice Pull Request WGHATickets

**What does your instant answer do?**
Displays Tickets to Partys and events in Germany. The spice is only triggered if request comes from Germany.

**What problem does your instant answer solve (Why is it better than organic links)?**
Directly display tickets for events inline when triggered by the right words.

**What is the data source for your instant answer? (Provide a link if possible)**
http://www.wasgehtheuteab.de

**Why did you choose this data source?**
I'm running the website

**Are there any other alternative (better) data sources?**
For German events no since other pages do not publish their data via an api

**What are some example queries that trigger this instant answer?**
Abendkasse Sarah
Tickets für Max Mutzke

**Which communities will this instant answer be especially useful for? (gamers, book lovers, etc)**
German party crowd

**Is this instant answer connected to a DuckDuckHack [instant answer idea](https://duck.co/ideas)?**
No

**Which existing instant answers will this one supersede/overlap with?**
Overlaps with SeatGeek

**Are you having any problems? Do you need our help with anything?**
No besides you need an api key which I can mail.

**\*\*Note:** Please attach a screenshot for new instant answer pull requests, and for pull requests which modify the look/design of existing instant answers.

## Checklist
Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information
[X] Wrote test file and added to t/ directory
[X] Verified that instant answer adheres to design guidelines(https://github.com/duckduckgo/duckduckgo-documentation/blob/master/duckduckhack/styleguides/design_styleguide.md)
[X] Tested cross-browser compatability

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [X] Google Chrome   
        [X] Firefox         
        [] Opera           
        [X] IE 10           

    - Windows 7
        [X] Google Chrome   
        [X] Firefox         
        [] Opera           
        [] IE 8            
        [X] IE 9            
        [X] IE 10           

    - Windows XP
        [] IE 7            
        [] IE 8            

    - Mac OSX
        [X] Google Chrome   
        [X] Firefox         
        [] Opera           
        [X] Safari          

    - iOS (iPhone)
        [X] Safari Mobile   
        [] Google Chrome   
        [] Opera           

    - iOS (iPad)
        [X] Safari Mobile   
        [] Google Chrome   
        [] Opera            

    - Android
        [X] Firefox         
        [] Native Browser  
        [X] Google Chrome   
        [] Opera
```

-----
IA Page: https://duck.co/ia/view/wghaticktes

### CODE REVIEW

**Comment by bigcurl** [comment]
bildschirmfoto 2015-09-18 um 21 32 53

Design is the same as spice wgha.

**Comment by moollaza** [comment]
@bigcurl thanks for this! I'll need to discuss this with the team before we can move it along.

Can you explain more about your data source? Is there any reason the current events Instant Answer doesn't work?

**Comment by bigcurl** [comment]
@moollaza For the site wgha.de I have tickets on sales for events. The tickets are provided by local (german) ticket vendors. The current IA event provider (with exception of wgha) are focused on events in America, the market local for them. They do not provide optimal results for concerts in Germany. The addition of the tickets provided by wgha should close this gap. 

To show the difference please see the attached screenshots showing search results for SeatGeek and WGHATickets.

Triggering SeatGeek with a query for artist in german location.
seat-geek-2

Same trigger with result from wghatickets:
wgha-result-2

**Comment by bigcurl** [comment]
Pushed the wrong button. Pull Request should not have been closed.

**Comment by moollaza** [comment]
@bigcurl thanks for explaining things! I'll get this deployed to the Beta server for further review :+1:

**Comment by bigcurl** [comment]
@moollaza Anything missing on this PR from my side?

**Comment by bigcurl** [comment]
@moollaza Let me know were I should send the secret key for the API.

**Comment by moollaza** [comment]
@bigcurl sorry for the delay. Please send the API key to moollaza@duckduckgo.com

**Comment by bigcurl** [comment]
@moollaza Ping. Whats with this one? Any showstoppers found?

File: lib/DDG/Spice/WGHATickets.pm [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+triggers any => @triggers;
+
+spice to => 'http://ddg.wasgehtheuteab.de/duckduckgo/tickets.json?q=$1&callback={{callback}}&api_key={{ENV{DDG_SPICE_WGHATICKETS_APIKEY}}}';
+
+spice proxy_cache_valid => "200 30m";
+
+handle query_lc => sub {
+  # This spice only triggers if request is from the Germany region
+  $a = $_ . ' ' . $loc->city;
```
Comment (moollaza) [inline lib/DDG/Spice/WGHATickets.pm]: @bigcurl we need to ensure we have a remainder so we don't just trigger on trigger words, e.g `tickets`

File: share/spice/wghatickets/wghatickets.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        if (!(api_result && api_result.length)) {
+            return Spice.failed('wghatickets');
+        }
+
+        var query = DDG.get_query();
+        var sourceUrl = 'http://www.wasgehtheuteab.de';
+
+        Spice.add({
+            id: 'wghaticktes',
```
Comment (moollaza) [inline share/spice/wghatickets/wghatickets.js]: typo here: `wghatickets`

**Comment by moollaza** [comment]
@bigcurl I've deployed the current code to beta. I'm worried by the relevancy of the results though:

https://beta.duckduckgo.com/?q=Tickets+f%C3%BCr+adele&ia=tickets
https://beta.duckduckgo.com/?q=Tickets+f%C3%BCr+ed+sheeran&ia=tickets

Don't seem to have anything relevant.

Our existing concert IA seems to handle it better:

- https://beta.duckduckgo.com/?q=adele+concert&ia=tickets
- https://beta.duckduckgo.com/?q=ed+sheeran+concert&ia=tickets

**Comment by bigcurl** [comment]
@moollaza I'll have a look at it after christmas break. Results have been better before. I'll investigate.

**Comment by moollaza** [comment]
> I'll have a look at it after christmas break. Results have been better before. I'll investigate.

@bigcurl thanks a lot, enjoy your break! :christmas_tree: :snowman:

**Comment by jdorweiler** [comment]
This pull request is being closed due to inactivity but, if anyone would like to continue it, please @ mention 'duckduckhack-contributors' for help and http://docs.duckduckhack.com/submitting/submitting-overview.html for reference.

**Pull request closed without merging**